### PR TITLE
Unit Testing with Jest Coverage

### PR DESCRIPTION
**Pull request by tyamadaaa**

### CODE REVIEW

**Comment by booqed** [comment]
@tyamadaaa here are the things to change:
1. add node_modules/ to .gitignore 
- you never commit node_modules to the repo
2. Don't separate a single test suite in a separate project -> first-test and second-test should be just a different tests suites in a single ```___test___``` folder.
3. Don't name your tests suites like that (i.e. ```first-test | second-test```) - names like that don't tell anything about what you are testing there. It should be something like you already have there ```functions.test.js``` for the second-test -> App.test.js name makes no sense, cuz you are not testing the App, meaning your "app" there is actually a module so your 'App.js' should just import module with those functions and you test that module separately

**Comment by booqed** [comment]
So pls put everything under a single project structure but separate files and please follow naming conventions.

**Comment by booqed** [comment]
and util you resolve .gitignore issue -> I cannot comment on the actual code, cuz there's 8,402 files in this pull request and it's not impossible to scroll to the actual code.

read about ```.gitignore``` and hot to add ```node_modules/``` there

File: __tests__/functions.test.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    test('should account for case', () => {
+        const names = ['Kate', 'Ruth', 'Wendy', 'kate'];
+        expect(functions.removeKNames(names)).not.toContain('Kate');
+        expect(functions.removeKNames(names)).not.toContain('kate');
+    });
+});
+});
+
+
```
Comment (keblodev) [inline __tests__/functions.test.js]: remove new lines

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const axios = require('axios');
+
+const functions = {
+    add: (x,y) => {
+        if(typeof x !== 'number' || typeof y !== 'number') {
+            return null;
+        } else {
+            return x + y;
+        }
+        },
```
Comment (keblodev) [inline functions.js]: indentation is off

File: README.md [hunk 3]
```diff
@@ -1 +1,22 @@
...
+
+- nvm latest version
+
+- Jest version 23.2.0
+
+## Set Up
+```
+$ npm init
+$ npm install --save-dev jest 
+```
```
Comment (keblodev) [inline README.md]: just ```npm i``` here

File: __tests__/functions.test.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    });
+
+    test('Using async wait to get Leanne Graham', async() => {
+        expect.assertions(1);
+        const data = await functions.fetchUser();
+        expect(data.name).toEqual('Leanne Graham');
+    });
+
+//Testing removeKNames function in various situation
+describe('remove names start with K', () => {
```
Comment (keblodev) [inline __tests__/functions.test.js]: Is this whole ```describe``` block is inside of another ```describe``` block testing different function? please move it outside as all tests before

File: functions.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    mult: (num1,num2) => num1 * num2,
+    createUser: () => {
+        const user = { firstName: 'Mike' }
+        user['lastName'] = 'Wood';
+        return user
+    },
+    fetchUser: () => axios.get('https://jsonplaceholder.typicode.com/users/1')
+        .then(res => res.data)
+        .catch(err => 'error'),
```
Comment (keblodev) [inline functions.js]: throw Error() or console.error()

**Comment by keblodev** [comment]
@tyamadaaa please apply requested changes and you're good to go

File: functions.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    mult: (num1,num2) => num1 * num2,
+    createUser: () => {
+        const user = { firstName: 'Mike' }
+        user['lastName'] = 'Wood';
+        return user
+    },
+    fetchUser: () => axios.get('https://jsonplaceholder.typicode.com/users/1')
+        .then(res => res.data)
+        .catch((error) => {
+            console.log(error);
```
Comment (keblodev) [inline functions.js:20]: fyi: you don't test this code, but that's ok for this test project. In a real one I would not allow that

**Comment by keblodev** [comment]
@tyamadaaa left a small code there, but you are good to merge this code.

**Pull request merged**